### PR TITLE
[DROOLS-5518] DMN strongly typed class compile errors for capitalized…

### PR DESCRIPTION
**Pull request by tkobayas**
…/non-capitalized properties conflict

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins retest this please

**Comment by mbiarnes** [comment]
Linux run full downstream build

**Comment by mbiarnes** [comment]
Linux execute compile downstream build

**Comment by mbiarnes** [comment]
Linux execute compile downstream build

**Comment by mbiarnes** [comment]
Jenkins do compile downstream build

**Comment by tkobayas** [comment]
As suggested by Matteo, I tried the simple approach:

> instead of playing with the first letter capitalization, we leave as-is

https://github.com/kiegroup/drools/pull/2993#discussion_r456937329

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java [hunk 1]
```diff
@@ -188,7 +186,7 @@ private boolean propertyTypePlaceHolder(Object n) {
 
     @Override
     public Optional<String> overriddenGetterName() {
-        String value = "get" + CodegenStringUtil.escapeIdentifier(ucFirst(originalMapKey));
+        String value = "get" + fieldName; // don't capitalize DROOLS-5518
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java]: ...while this change is fine and can remain as-is 👍

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java [hunk 2]
```diff
@@ -197,7 +195,7 @@ private boolean propertyTypePlaceHolder(Object n) {
 
     @Override
     public Optional<String> overriddenSetterName() {
-        return Optional.of("set" + CodegenStringUtil.escapeIdentifier(ucFirst(originalMapKey)));
+        return Optional.of("set" + fieldName); // don't capitalize DROOLS-5518
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java]: ...also this change is fine and can remain as-is

**Review by tarilabs — COMMENTED** [review]
This is fine but I believe we can actually do a little better, reviewing our original conversations.
The premise is that I would normally assume authors don't create DMN model with node name differing only in capitalisation, in other words I would assume people don't usually create two InputData such as `myPerson` and `MyPerson` in the same model.
So that said, we could plug the Strategy in the constructor of `DMNDeclaredField` depending on conflict analysis made on the _caller_ code.
wdyt?

**Review by hellowdan — APPROVED** [review]

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
> we could plug the Strategy in the constructor of DMNDeclaredField depending on conflict analysis made on the caller code.

Thank you, @tarilabs . I agree with the idea and introduced `FieldGenStrategy`. Does this look good?

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/FieldGenStrategy.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package org.kie.dmn.typesafe;
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/FieldGenStrategy.java:17]: please don't forget license header

**Review by tarilabs — APPROVED** [review]
Looking good, thanks for the update!

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/FieldGenStrategy.java (hunk 3 above)
Comment (tkobayas) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/FieldGenStrategy.java:17]: Oops, thanks!

**Comment by tkobayas** [comment]
Jenkins retest this

**Pull request merged**